volpe/annotate_utils.py

```diff
@@ -39,6 +39,50 @@ def math_assign(self, tree: TypeTree):
     return self.visit(tree)
 
 
+def chain_comp(self, tree: TypeTree):
+    # Single comparison
+    if len(tree.children) == 3:
+        a, symbol, b = tree.children[0], tree.children[1], tree.children[2]
+        tree.data = symbol_to_data(symbol)
+        tree.children = [a, b]
+        self.visit_children(tree)
+
+    # Chained comparison
+    else:
+        # Separate expressions and comparison operators.
+        expressions = tree.children[::2]
+        comparisons = tree.children[1::2]
+
+        # Generate comparison trees.
+        comp_trees = [
+            TypeTree(symbol_to_data(symbol), [a, b], tree.meta)
+            for symbol, a, b in zip(comparisons, expressions[:-1], expressions[1:])
+        ]
+
+        # Build up nested tree.
+        prev_tree = TypeTree("logic_and", [comp_trees[0], comp_trees[1]], tree.meta)
+        for comp_tree in comp_trees[2:]:
+            prev_tree = TypeTree("logic_and", [prev_tree, comp_tree], tree.meta)
+
+        # Override this node with last.
+        tree.data = prev_tree.data
+        tree.children = prev_tree.children
+
+        self.visit_children(tree)
+
+
+def symbol_to_data(symbol):
+    if symbol == "<=":
+        return "less_equals"
+    if symbol == ">=":
+        return "greater_equals"
+    if symbol == "<":
+        return "less"
+    if symbol == ">":
+        return "greater"
```

Comment:
it would be better to use a dictionary here

Reply:
yes, good point